.github/workflows/pull-request-ci.yaml

```diff
@@ -0,0 +1,307 @@
+name: CI
+on: push
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    env:
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+    defaults:
+      run:
+        working-directory: kubecf
+    steps:
+    - name: Checkout KubeCF
+      uses: actions/checkout@v2
+      with:
+        path: kubecf
+        fetch-depth: 0
+        submodules: recursive
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+
+    # Run the linters; split them out so it's easier to identify which one failed
+    - run: make shellcheck
+    - run: make yamllint
+    - run: make helmlint
+
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    needs: lint
+    env:
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+    defaults:
+      run:
+        working-directory: kubecf
+    steps:
+    - name: Checkout KubeCF
+      uses: actions/checkout@v2
+      with:
+        path: kubecf
+        fetch-depth: 0
+        submodules: recursive
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+    - run: make kubecf-bundle
+      env:
+        TARGET_FILE: ${{ github.workspace }}/kubecf-bundle.tgz
+    - run: |
+        mkdir bundle
+        tar xf kubecf-bundle.tgz -C bundle
+      working-directory: ${{ github.workspace }}
+    - name: Upload bundle
+      uses: actions/upload-artifact@v2
+      with:
+        name: kubecf-bundle.tgz
+        path: ${{ github.workspace }}/bundle
+
+  tests:
+    name: tests
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    needs: build
+    env:
+      # For KubeCF
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+      FEATURE_AUTOSCALER: "true"
+      # For catapult
+      # Note that credentials-related parts are in the individual steps
+      BACKEND: gke
+      DOWNLOAD_CATAPULT_DEPS: "false"
+
+    defaults:
+      run:
+        working-directory: kubecf
+
+    strategy:
+      matrix:
+        include:
+        - backend: diego
+          experimental: false
+        - backend: eirini
+          experimental: true
+
+    steps:
+    - name: Checkout KubeCF
```

Comment:
Can you please leave one empty row between each step?
If warranted even a comment helps read the yaml.

e.g.

```yaml

    # We need catapult to create Kubernetes clusters
    - name: Checkout catapult
      uses: actions/checkout@v2
      with:
        repository: SUSE/catapult
        path: catapult

    # We'll be caching foo and bar
    - name: cache.tools
      uses: actions/cache@v2
      with:
        path: ${{ github.workspace }}/tools
        key: ${{ runner.os }}-tools
```